stellar-auth/pkg/utils/password_validation.go

```diff
@@ -83,10 +111,28 @@ func ValidatePassword(input string) error {
 	if len(invalidCharacteres) > 0 {
 		failedValidations["invalid character"] = fmt.Sprintf("password cannot contain any invalid characters ('%s')", strings.Join(invalidCharacteres, "', '"))
 	}
+	if !hasLength {
+		failedValidations["length"] = fmt.Sprintf("password length must be between %d and %d characters", passwordMinLength, passwordMaxLength)
+	}
 
 	if len(failedValidations) == 0 {
-		return nil
+		if pv.determineIfCommonPassword(input) {
+			failedValidations["common password"] = "password is determined to be too common"
+		} else {
+			return nil
+		}
 	}
 
 	return &ValidatePasswordError{FailedValidationsMap: failedValidations}
 }
+
+func (pv *PasswordValidator) determineIfCommonPassword(input string) bool {
+	input = strings.ToLower(input)
+	for _, commonPassword := range pv.commonPasswordsList {
+		if input == commonPassword {
```

Comment:
This linear search can be quite slow depending on how long the list is. You may want to change the datastructure for `commonPasswordsList` to a map.
```go
_, found := pv.commonPasswordsMap[input]
return found
```

Reply:
We'd have to change `GetPasswordValidatorInstance` to create a map instead of a list. 
```go
passwordsMap := make(map[string]struct{})
for _, password := range passwordsList {
    cleanedPassword := strings.TrimSpace(strings.ToLower(password)) // optionally trimming and lowercasing in case it's needed. 
    passwordsMap[cleanedPassword] = struct{}{}
}
```

Reply:
ok, having instant lookup would be a great improvement!

Reply:
this is great @ziyliu